sample_consensus/include/pcl/sample_consensus/impl/ransac.hpp

```diff
@@ -200,7 +198,21 @@ pcl::RandomSampleConsensus<PointT>::computeModel (int)
         break;
       }
     } // while
-  } // omp parallel
+  };
+
+  int threads = threads_ == 0 ? omp_get_num_threads() : threads_;
+
+  if (executor::is_instance_of_base_v<executor::omp_executor, Executor>) {
+    exec.execute([&](){
+    #pragma omp parallel if(threads > 0) num_threads(threads) shared(k, skipped_count, n_best_inliers_count) firstprivate(selection, model_coefficients) // would be nice to have a default(none)-clause here, but then some compilers complain about the shared const variables
```

Comment:
Should the pragma go in the executor's execute (with threads as shape) and the lambda have the variable references?